### PR TITLE
Fixed html renderer

### PR DESCRIPTION
**Pull request by foxinushka**

### CODE REVIEW

**Comment by dinhviethoa** [comment]
Could you describe the issue you had?

File: src/core/renderer/MCHTMLRenderer.cpp [hunk 1]
```diff
@@ -464,8 +464,9 @@ static String * htmlForAbstractMultipartMixed(AbstractMultipart * part, htmlRend
         AbstractPart * subpart = (AbstractPart *) part->parts()->objectAtIndex(i);
         String * substring = htmlForAbstractPart(subpart, context);
         if (context->pass != 0) {
-            if (substring == NULL)
-                return NULL;
+            if (substring == NULL) {
```
Comment (CodaFi) [inline src/core/renderer/MCHTMLRenderer.cpp]: If you're getting null here, it indicates a failure in `htmlForAbstractPart`.  I don't think this should be the workaround for that kind of logic error.

**Comment by foxinushka** [comment]
I found where I've mistaken.

I have a message with next structure:

```
multipart
   multipart
      text/plain
      text/html
   text/plain (inline attachment)
```

When I render HTML content of the message, I should skip text/plain (inline attachment) part.
I skip it in shouldShowPart method of MCOHTMLRendererDelegate. But the method isn't be called for text/... parts, and my code contained call of shouldShowPart in dataForIMAPPart. So, It is my mistake. 

Please, check another fix. 
Thanks!

**Comment by dinhviethoa** [comment]
I'm closing that issue then.

**Pull request closed without merging**